sopel/modules/currency.py

```diff
@@ -1,102 +1,189 @@
+    amount = others.pop(0)
+    of = others.pop(0)
+    others.pop(0)
+
+#    amount, of, _, *others = query.split() # I'd much rather use this, but it's not python 2.7 compatible
+
     try:
         amount = float(amount)
     except ValueError:
         bot.reply("Sorry, I didn't understand the input.")
     except OverflowError:
         bot.reply("Sorry, input amount was out of range.")
-    display(bot, amount, of, to)
+
+    out_string = '{} {} is'.format(amount, of.upper())
+    for to in others:
+        try:
+            out_string = build_reply(bot, amount, of.upper(), to.upper(), out_string)
+        except (requests.exceptions.HTTPError, requests.exceptions.ConnectionError) as e:
+            bot.reply("Something went wrong while I was getting the exchange rate.")
+            stderr("Error in GET request: {}".format(e))
+            return NOLIMIT
+        except ValueError:
+            return NOLIMIT
+
+    bot.reply(out_string[0:-1])
+
+
+@commands('cur', 'currency', 'exchange')  # this
+@example('.cur 100 usd in btc cad eur')
+def exchange_cmd(bot, trigger):
+    if not trigger.group(2):
+        return bot.reply("No search term. An example: .cur 100 usd in btc cad eur")
+
+    match = regex.match(trigger.group(2))
+    exchange(bot, match)
+
+
+@rule(regex)
+@example('100 usd in btc cad eur')
+def exchange_re(bot, trigger):
+    if bot.config.currency.enable_regex:
+        match = regex.match(trigger)
+        exchange(bot, match)
 
 
-def display(bot, amount, of, to):
+def build_reply(bot, amount, of, to, out_string):
     if not amount:
         bot.reply("Zero is zero, no matter what country you're in.")
+
+    rate_raw = ''
     try:
-        of_rate, of_name = get_rate(of)
-        if not of_name:
-            bot.reply("Unknown currency: %s" % of)
-            return
-        to_rate, to_name = get_rate(to)
-        if not to_name:
-            bot.reply("Unknown currency: %s" % to)
-            return
-    except Exception:  # TODO: Be specific
-        bot.reply("Something went wrong while I was getting the exchange rate.")
-        return NOLIMIT
+        rate_raw = get_rate(of, to)
+        rate = float(rate_raw)
+    except ValueError:
+        bot.reply(rate_raw)
+        raise
 
-    result = amount / of_rate * to_rate
-    bot.say("{:.2f} {} ({}) = {:.2f} {} ({})".format(amount, of.upper(), of_name,
-                                             result, to.upper(), to_name))
+    result = float(rate * amount)
 
+    if to == 'BTC':
+        return out_string + ' {:.5f} {},'.format(result, to)
 
-@commands('btc', 'bitcoin')
-@example('.btc 20 EUR')
-def bitcoin(bot, trigger):
-    # if 2 args, 1st is number and 2nd is currency. If 1 arg, it's either the number or the currency.
-    to = trigger.group(4)
-    amount = trigger.group(3)
-    if not to:
-        to = trigger.group(3) or 'USD'
-        amount = 1
+    return out_string + ' {:.2f} {},'.format(result, to)
+
+
+class CurrencySection(StaticSection):
```

Comment:
Most modules seem to define the config at the top. Can this be moved up top for consistency since it is a built-in plugin?

Reply:
No reason it can't be. Shouldn't cause any `NameError`s (unlike when I tried to move the `help` config stuff higher in #1451).

The `CurrencySection` class, `configure`, and `setup` should all be moved up to just before the first `def` (`btc_rate()`).

Reply:
Done :)